cartoframes/dataset.py

```diff
@@ -39,13 +39,16 @@ class Dataset(object):
     DEFAULT_RETRY_TIMES = 3
 
     def __init__(self, table_name=None, schema='public', query=None, df=None, gdf=None, state=None, context=None):
-        # TODO: error control (https://github.com/CartoDB/cartoframes/issues/669)
-
         self.table_name = normalize_name(table_name)
         self.schema = schema
         self.query = query
         self.df = df
         self.gdf = gdf
+
+        if not self._validate_init():
+            raise CartoException('Wrong Dataset creation. You should use one of the class methods:'
```

Comment:
To me this is more of a `ValueError` (or something like that) instead of a `CartoException`.

I also propose changing the error text to: `Invalid Dataset inputs. Specify only one of 'table_name', 'query', 'df', or 'gdf'. Or use one of the following class methods: from_table, from_query, from_dataframe, from_geodataframe, from_geojson.`

Reply:
:+1:  to `ValueError`

About the message, we don't want the users use the class properties, but the class methods.